datadog_checks_base/datadog_checks/base/config.py

```diff
@@ -7,13 +7,16 @@ def is_affirmative(value):
     """
     Attempt to convert different type of values to a meaningful boolean
     """
-    # try string cast
-    if isinstance(value, str):
-        return value.lower() in {'yes', 'true', '1', 'y', 'on'}
-
+    # uses an explicit set of all cases lower and upper rather than value.lower() in {'yes', 'true', '1', 'y', 'on'}
     # use object's implementation of `__bool__`, it's faster than cast.
     # None -> False, 0 -> False, 1 -> True, etc.
-    return not not value
+    return value in {
+        '1',
+        'ON', 'On', 'oN', 'on',
+        'TRUE', 'TRUe', 'TRuE', 'TRue', 'TrUE', 'TrUe', 'TruE', 'True',
+        'tRUE', 'tRUe', 'tRuE', 'tRue', 'trUE', 'trUe', 'truE', 'true',
+        'Y', 'y',
+        'YES', 'YEs', 'YeS', 'Yes', 'yES', 'yEs', 'yeS', 'yes'} or not not value
```

Comment:
```python
def is_affirmative(value):
    """
    Attempt to convert different type of values to a meaningful boolean
    """
    # try string cast
    if isinstance(value, str):
        return value.lower() in {'yes', 'true', '1', 'y', 'on'}

    # use object's implementation of `__bool__`, it's faster than cast.
    # None -> False, 0 -> False, 1 -> True, etc.
    return not not value


def is_affirmative2(value):
    """
    Attempt to convert different type of values to a meaningful boolean
    """
    # use object's implementation of `__bool__`, it's faster than cast.
    # None -> False, 0 -> False, 1 -> True, etc.
    return value in {
        '1',
        'ON', 'On', 'oN', 'on',
        'TRUE', 'TRUe', 'TRuE', 'TRue', 'TrUE', 'TrUe', 'TruE', 'True',
        'tRUE', 'tRUe', 'tRuE', 'tRue', 'trUE', 'trUe', 'truE', 'true',
        'Y', 'y',
        'YES', 'YEs', 'YeS', 'Yes', 'yES', 'yEs', 'yeS', 'yes'} or not not value


if __name__ == '__main__':
    from timeit import default_timer as timer

    for fct in [is_affirmative, is_affirmative2]:
        s = timer()
        for t in ['Yes', False, True, 0, 1, 'DWIJDWOD'] * 10000:
            fct(t)
        print(str(timer() - s))
```

> 0.014073899999999997
> 0.008309300000000006

Reply:
Hallo @Jallasia are there any performance issues you are experiencing with the agent? I do not think the bottleneck for performance in the agent is here (compared to making REST calls as many integrations do) so the performance gain on this method won't really have much effect on the overall agent performance, so I don't think the readability drop is justified. Also having a list of values is more bug prone than the current implementation.

Reply:
Well if you prefer we can do it like that:

```python
def capitalization_permutations(value):
    """Generates the different ways of capitalizing the letters in
    the string s.

    >>> list(capitalization_permutations('abc'))
    ['ABC', 'aBC', 'AbC', 'abC', 'ABc', 'aBc', 'Abc', 'abc']
    >>> list(capitalization_permutations(''))
    ['']
    >>> list(capitalization_permutations('X*Y'))
    ['X*Y', 'x*Y', 'X*y', 'x*y']
    """
    if value == '':
        yield ''
        return
    for rest in capitalization_permutations(value[1:]):
        yield value[0].upper() + rest
        if value[0].upper() != value[0].lower():
            yield value[0].lower() + rest


def capitalization_permutations_set(*value):
    return {p for v in value for p in capitalization_permutations(v)}

IS_TRUE_SET = capitalization_permutations_set('yes', 'true', '1', 'y', 'on')

def is_affirmative(value):
    """
    Attempt to convert different type of values to a meaningful boolean
    """
    # use object's implementation of `__bool__`, it's faster than cast.
    # None -> False, 0 -> False, 1 -> True, etc.
    return value in IS_TRUE_SET or (not not value and not isinstance(value, str))
```

That still leave a little computing time at startup but it's more clear to read

Reply:
You are right, i will split that is another PR anyway, that's better